metadata-db-app/src/migrations/055_setup_services_table.clj

```diff
@@ -0,0 +1,58 @@
+(ns migrations.055-setup-services-table
+  (:require
+   [config.mdb-migrate-helper :as h]))
+
+(def ^:private services-column-sql
+  "id NUMBER,
+  concept_id VARCHAR(255) NOT NULL,
+  native_id VARCHAR(1030) NOT NULL,
+  metadata BLOB NOT NULL,
+  format VARCHAR(255) NOT NULL,
+  revision_id INTEGER DEFAULT 1 NOT NULL,
+  revision_date TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL,
+  created_at TIMESTAMP WITH TIME ZONE DEFAULT SYSTIMESTAMP NOT NULL,
+  deleted INTEGER DEFAULT 0 NOT NULL,
+  user_id VARCHAR(30),
+  service_name VARCHAR(20),
+  transaction_id INTEGER DEFAULT 0 NOT NULL")
+
+(def ^:private services-constraint-sql
+  (str "CONSTRAINT services_pk PRIMARY KEY (id), "
+       ;; Unique constraint on native id and revision id
+       "CONSTRAINT services_con_rev UNIQUE (native_id, revision_id)
+       USING INDEX (create unique index services_ucr_i ON cmr_services (native_id, revision_id)), "
+
+       ;; Unique constraint on concept id and revision id
+       "CONSTRAINT services_cid_rev UNIQUE (concept_id, revision_id)
+       USING INDEX (create unique index services_cri ON cmr_services (concept_id, revision_id))"))
+
+(defn- create-services-table
+  []
+  (h/sql
+   (format "CREATE TABLE METADATA_DB.cmr_services (%s, %s)"
+           services-column-sql services-constraint-sql)))
+
+
+(defn- create-services-indices
+  []
+  (h/sql "CREATE INDEX services_crdi ON METADATA_DB.cmr_services (concept_id, revision_id, deleted)")
```

Comment:
I think the indexes apply to joins and where clause, not he selects. In other words, it is not necessary to create index on concept_id and service_name to speed up "select concept_id from cmr_services where service_name='foo'". Am I wrong? If we select a lot of fields out of the table, the query engine might eventually choose a table scan rather than using the index, but that is a different issue.